src/Compilers/CSharp/Test/Semantic/Semantics/TypeParameterCreationTests.cs

```diff
@@ -0,0 +1,272 @@
+
+        var symbolInfo = model.GetSymbolInfo(objectCreationNode);
+        Assert.True(symbolInfo.IsEmpty);
+
+        var typeNode = objectCreationNode.Type;
+
+        typeInfo = model.GetTypeInfo(typeNode);
+        Assert.Null(typeInfo.Type);
+        Assert.Null(typeInfo.ConvertedType);
+
+        symbolInfo = model.GetSymbolInfo(typeNode);
+        Assert.Equal("T", symbolInfo.Symbol?.ToTestDisplayString());
+    }
+
+    [Theory]
+    [InlineData("new()")]
+    [InlineData("class, new()")]
+    [InlineData("struct")]
+    public void CreateGenericTypeParameterObject_ImplicitCreation(string constraint)
+    {
+        var source = $$"""
+            class C
+            {
+                void M<T>() where T : {{constraint}}
+                {
+                    T t = new();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+
+        var objectCreationNode = tree
+            .GetCompilationUnitRoot()
+            .DescendantNodes()
+            .OfType<ImplicitObjectCreationExpressionSyntax>()
+            .Single();
+
+        var typeInfo = model.GetTypeInfo(objectCreationNode);
+        Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+        Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+        var symbolInfo = model.GetSymbolInfo(objectCreationNode);
+        Assert.True(symbolInfo.IsEmpty);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("where T : class")]
+    public void CreateGenericTypeParameterObject_ExplicitCreation_ErrorRecovery_NoNewConstraint(string constraintClause)
+    {
+        var source = $$"""
+            class C
+            {
+                void M<T>() {{constraintClause}}
+                {
+                    T t = new T();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (5,15): error CS0304: Cannot create an instance of the variable type 'T' because it does not have the new() constraint
+            //         T t = new T();
+            Diagnostic(ErrorCode.ERR_NoNewTyvar, "new T()").WithArguments("T").WithLocation(5, 15));
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+
+        var objectCreationNode = tree
+            .GetCompilationUnitRoot()
+            .DescendantNodes()
+            .OfType<ObjectCreationExpressionSyntax>()
+            .Single();
+
+        var typeInfo = model.GetTypeInfo(objectCreationNode);
+        Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+        Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+        var symbolInfo = model.GetSymbolInfo(objectCreationNode);
+        Assert.True(symbolInfo.IsEmpty);
```

Comment:
So the PR only changes this particular scenario: if we get symbol info from the whole `new T()` (or just `new()`) we get empty symbol info both in legal and error cases